app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -934,16 +1116,227 @@ class AppointmentRepositoryAndroidTest {
+        .blockingFirst()
+        .associateBy({ it.appointment.uuid }, { it.patientLastSeen })
+
+    // then
+    val expected = mapOf(
+        appointmentUuidForFirstPatient to laterRecordedBloodSugarForFirstPatient.recordedAt
+    )
+    assertThat(bloodSugarByAppointmentUuid).isEqualTo(expected)
+  }
+
+  @Test
+  fun when_picking_overdue_appointment_and_blood_pressure_is_latest_compared_to_blood_sugar_then_blood_pressure_should_be_considered() {
+    fun createBloodSugar(patientProfile: PatientProfile, recordedAt: Instant): BloodSugarMeasurement {
+      return testData.bloodSugarMeasurement(
+          patientUuid = patientProfile.patient.uuid,
+          recordedAt = recordedAt
+      )
+    }
+
+    fun createBloodPressure(patientProfile: PatientProfile, recordedAt: Instant): BloodPressureMeasurement {
+      return testData.bloodPressureMeasurement(
+          patientUuid = patientProfile.patient.uuid,
+          recordedAt = recordedAt
+      )
+    }
+
+    fun scheduleAppointment(
+        appointmentUuid: UUID,
+        patientProfile: PatientProfile
+    ): Single<Appointment> {
+      return appointmentRepository.schedule(
+          patientUuid = patientProfile.patient.uuid,
+          appointmentUuid = appointmentUuid,
+          appointmentDate = LocalDate.parse("2017-12-30"),
+          appointmentType = Manual,
+          appointmentFacilityUuid = facility.uuid,
+          creationFacilityUuid = facility.uuid
+      )
+    }
+
+    // given
+    val firstPatient = testData.patientProfile(
+        patientUuid = UUID.fromString("e1943cfb-faf0-42c4-b5b6-14b5153295b2"),
+        generatePhoneNumber = true
+    )
+
+    patientRepository.save(listOf(firstPatient)).blockingAwait()
+
+    val earlierRecordedBloodSugarForFirstPatient = createBloodSugar(
+        patientProfile = firstPatient,
+        recordedAt = Instant.parse("2017-12-31T23:59:59Z")
+    )
+    val laterRecordedBPForFirstPatient = createBloodPressure(
+        patientProfile = firstPatient,
+        recordedAt = Instant.parse("2018-01-01T00:00:00Z")
+    )
+
+    bloodSugarRepository.save(listOf(earlierRecordedBloodSugarForFirstPatient))
+    bpRepository.save(listOf(laterRecordedBPForFirstPatient)).blockingAwait()
+
+    val appointmentUuidForFirstPatient = UUID.fromString("d9fd734d-13b8-43e3-a2d7-b40341699050")
+
+    scheduleAppointment(appointmentUuidForFirstPatient, firstPatient).blockingGet()
+    scheduleAppointment(appointmentUuidForFirstPatient, firstPatient).blockingGet()
+
+    // when
+    val bloodSugarByAppointmentUuid = appointmentRepository
+        .overdueAppointments(since = LocalDate.now(clock), facility = facility)
+        .blockingFirst()
+        .associateBy({ it.appointment.uuid }, { it.patientLastSeen })
+
+    // then
+    val expected = mapOf(
+        appointmentUuidForFirstPatient to laterRecordedBPForFirstPatient.recordedAt
+    )
+    assertThat(bloodSugarByAppointmentUuid).isEqualTo(expected)
+  }
```

Comment:
I think we can drop the "First" qualifier from the variable names since we only have one patient.